src/tree/SubscriptionTreeItem.ts

```diff
@@ -21,46 +21,67 @@ import { GroupTreeItemBase } from './GroupTreeItemBase';
 import { ResourceGroupTreeItem } from './ResourceGroupTreeItem';
 
 export class SubscriptionTreeItem extends SubscriptionTreeItemBase {
+
+    private async getResourceGroups(context: IActionContext): Promise<ResourceGroup[]> {
+        if (!this.cache.resourceGroups.length || !this._keepCache) {
+            const client: ResourceManagementClient = await createResourceClient([context, this.subscription]);
+            this.cache.resourceGroups = await uiUtils.listAllIterator(client.resourceGroups.list());
+        }
+        return this.cache.resourceGroups;
+    }
+
     public readonly childTypeLabel: string = localize('resourceGroup', 'Resource Group');
 
-    private _nextLink: string | undefined;
-    private _items: GroupableResource[] = [];
+    private cache: {
+        resourceGroups: ResourceGroup[];
+        nextLink?: string;
+        appResources: AppResource[];
+        groupableResources: GroupableResource[];
+    }
     private _treeMap: { [key: string]: GroupTreeItemBase } = {};
 
-    private rgsItem: AppResource[] = [];
+    private resetCache(): void {
+        this.cache = {
+            resourceGroups: [],
+            appResources: [],
```

Comment:
`AppResource` and `GroupableResource` are meant to overlap; i.e. the relevant object implements both of them.